.github/workflows/main.yml

```diff
@@ -89,6 +89,8 @@ jobs:
       run: docker build -t prairielearn/prairielearn:${{ env.COMMIT_SHA }} .
       env: 
         PRAIRIELEARN_IMAGE_TAG: ${{ env.COMMIT_SHA }}
+    - name: Tag prairielearn image as latest
+      run docker tag prairielearn/prairielearn:${{ env.COMMIT_SHA }} prairielearn/prairielearn:latest
```

Comment:
🙃 This was the root cause of why CI was behaving strangely - we were testing `prairielearn/prairielearn:latest` from Docker Hub, not the one we just built!

Reply:
Oh dear! 🤦